apps/arena/lib/arena/game/player.ex

```diff
@@ -167,4 +200,13 @@ defmodule Arena.Game.Player do
   end
 
   defp maybe_trigger_natural_heal(player, _), do: player
+
+  defp add_or_remove_moving_action(current_actions, direction) do
+    if direction == {0.0, 0.0} do
+      current_actions -- [%{action: :MOVING, duration: 0}]
+    else
+      current_actions ++ [%{action: :MOVING, duration: 0}]
+    end
+    |> Enum.uniq()
+  end
```

Comment:
just a comment, you could match the direction directly on the declaration:
```
defp add_or_remove_moving_action(current_actions, {0.0, 0.0}), do: REMOVE
defp add_or_remove_moving_action(current_actions, _direction), do: APPEND
```

Reply:
If not a problem I'll address this in a follow up because this was copy&paste from `game_updater.ex` and doesn't really have anything to do with this PR

About the uniqueness, sounds reasonable. What we where trying to solve here was that without it the actions grow unbounded because they grow with each move action

Reply:
yes! lets address this later, it was just a comment about a possible code change